include/zephyr/fs/nvs.h

```diff
@@ -132,8 +132,13 @@ int nvs_delete(struct nvs_fs *fs, uint16_t id);
  *
  * @return Number of bytes read. On success, it will be equal to the number of bytes requested
  * to be read. When the return value is larger than the number of bytes requested to read this
- * indicates not all bytes were read, and more data is available. On error, returns negative
- * value of errno.h defined error codes.
+ * indicates not all bytes were read and more data is available.
+ *
+ * When @p len is 0, no data is copied into @p data. Instead, the function returns the full
+ * length of the entry, allowing the caller to determine the buffer size needed for a subsequent
```

Comment:
I wouldn't add this to the documentation. Although it is valid in many cases there is the possibility that another thread updates the data in between a first and a second call to nvs_read. As long as there is no update the proposed procedure can be used, however it is not guaranteed.